### PR TITLE
phpstan: some fixes

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- 1st commit ...
- 2nd/3rd call helper/model instead of wrapper method, that cant be resolved
- 4th added variable + DOCblocks that can be resolved

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Catalog/Block/Widget/Link.php [hunk 1]
```diff
@@ -68,18 +68,15 @@ public function getHref()
             $idPath = explode('/', $this->_getData('id_path'));
 
             if (isset($idPath[0]) && isset($idPath[1]) && $idPath[0] == 'product') {
-
-                /** @var Mage_Catalog_Helper_Product $helper */
-                $helper = $this->_getFactory()->getHelper('catalog/product');
+                $helper = Mage::helper('catalog/product');
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Block/Widget/Link.php:71]: Is this BC? Suppose there is an override of `_getFactory()` or `getHelper()`?

**Comment by sreichel** [comment]
@kiatng i'll revert mentioned lines, because change was not necessary, but one question remains ... is not using `getHelper($someName)` a BC breaking change?

Why would you override ...

```
    public function getHelper($helperClass)
    {
        return Mage::helper($helperClass);
    }
```

.. instead of rewriting the helper class itself?

**Comment by sreichel** [comment]
I'll split in in samller chuncks.

File: app/code/core/Mage/Catalog/Block/Widget/Link.php [hunk 2]
```diff
@@ -79,7 +78,7 @@ public function getHref()
                 $categoryId = $idPath[1];
                 if ($categoryId) {
                     /** @var Mage_Catalog_Helper_Category $helper */
-                    $helper = $this->_getFactory()->getHelper('catalog/category');
+                    $helper = $this->_getFactory()->getHelper('catalog/product');
```
Comment (fballiano) [inline app/code/core/Mage/Catalog/Block/Widget/Link.php:81]: was this correct to change the model from category to product?

Schermata 2022-08-19 alle 00 01 47

Comment (sreichel) [reply app/code/core/Mage/Catalog/Block/Widget/Link.php:81]: No, this was not correct. Mistake when reverting ... sorry.

**Comment by kiatng** [comment]
> Why would you override ...

I do no such thing. But we know there are all sorts of extensions, some of which do crazy things.

**Pull request closed without merging**